simulator/src/main/java/com/hazelcast/simulator/coordinator/Coordinator.java

```diff
@@ -130,332 +117,261 @@ private void initAgents() throws Exception {
+                        testSuite.id));
             }
+            LOGGER.info("    " + ip + " copied");
         }
+        LOGGER.info(format("Finished copying workerClasspath '%s' to agents", workerClassPath));
     }
 
-    private void runParallel() throws InterruptedException, java.util.concurrent.ExecutionException {
-        echo(format("Running %s tests parallel", testSuite.size()));
-
-        ExecutorService executor = Executors.newFixedThreadPool(testSuite.size());
-
-        final int maxTestCaseIdLength = getMaxTestCaseIdLength(testSuite.testCaseList);
-
-        List<Future> futures = new LinkedList<Future>();
-        for (final TestCase testCase : testSuite.testCaseList) {
-            Future f = executor.submit(new Runnable() {
-                @Override
-                public void run() {
-                    try {
-                        TestCaseRunner runner = new TestCaseRunner(testCase, testSuite, Coordinator.this, maxTestCaseIdLength);
-                        boolean success = runner.run();
-                        if (!success && testSuite.failFast) {
-                            LOGGER.info("Aborting testsuite due to failure");
-                            return;
-                        }
-
-//                       if (!success || workerJvmSettings.refreshJvm) {
-//                           terminateWorkers();
-//                           startWorkers();
-//                       }
-                    } catch (Exception e) {
-                        throw new RuntimeException(e);
-                    }
-                }
-            });
-
-            futures.add(f);
+    private void uploadYourKitIfNeeded() {
+        if (!"yourkit".equals(workerJvmSettings.profiler)) {
+            return;
         }
 
-        for (Future f : futures) {
-            f.get();
+        // TODO: in the future we'll only upload the requested YourKit library (32 or 64 bit)
+        LOGGER.info("Uploading YourKit dependencies to agents");
+        for (String ip : agentsClient.getPublicAddresses()) {
+            bash.ssh(ip, format("mkdir -p hazelcast-simulator-%s/yourkit", getSimulatorVersion()));
+
+            bash.execute(format("rsync --ignore-existing -avv -e \"ssh %s\" %s/yourkit %s@%s:hazelcast-simulator-%s/",
+                    props.get("SSH_OPTIONS", ""),
+                    getSimulatorHome().getAbsolutePath(),
+                    props.get("USER"),
+                    ip,
+                    getSimulatorVersion()));
         }
     }
 
-    private void terminateWorkers() throws Exception {
-        echo("Terminating workers");
-        agentsClient.terminateWorkers();
-        echo("All workers have been terminated");
-    }
-
-    private long startWorkers() throws Exception {
+    private void startWorkers() throws Exception {
         List<AgentMemberLayout> agentMemberLayouts = initMemberLayout();
 
-        long startMs = System.currentTimeMillis();
+        long started = System.nanoTime();
```

Comment:
hm, I guess in theory `System.currentTimeMillis()` is not guaranteed to be always monotonic (on multiple CPUs?). 

that's a fair point.